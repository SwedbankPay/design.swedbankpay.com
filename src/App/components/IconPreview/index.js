import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const IconPreview = ({ name, squaredFlag, preview, previewSize, size, className, type }) => {
    const classNames = className ? className.split(" ") : [];
    const iconClasses = classnames(
        type,
        size ? `${type}-${size}` : "",
        type === "material-icons" ? "" : `${type}-${name}`,
        squaredFlag ? "flag-icon-squared" : "",
        ...classNames
    );

    return preview || previewSize ? (
        <div className="icon-preview">
            <i className={iconClasses} aria-hidden="true">{iconClasses.includes("material-icons") ? name : null}</i>
            <code className="code-tags code-tags-secondary mt-2">
                {previewSize ? size : name}
            </code>
        </div>
    ) : <i className={iconClasses} aria-hidden="true">{iconClasses.includes("material-icons") ? name : null}</i>;
};

IconPreview.propTypes = {
    name: PropTypes.string.isRequired,
    type: PropTypes.oneOf([
        "material-icons",
        "payment-icon",
        "flag-icon"
    ]),
    squaredFlag: PropTypes.bool,
    preview: PropTypes.bool,
    size: PropTypes.oneOf([
        "tiny",
        "small",
        "medium",
        "large",
        "huge"
    ]),
    className: PropTypes.string
};

export default IconPreview;
