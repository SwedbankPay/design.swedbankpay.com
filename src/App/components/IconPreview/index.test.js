import React from "react";
import { shallow } from "enzyme";

import IconPreview from "./index";

describe("Component: IconPreview - ", () => {
    it("is defined", () => {
        expect(IconPreview).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<IconPreview name="amex" type="payment-icon" />);

        expect(wrapper).toMatchSnapshot();
    });

    it("prop name is marked as required", () => {
        console.error = jest.fn();

        const wrapper = shallow(<IconPreview type="payment-icon" />);

        expect(console.error).toHaveBeenCalled();
        expect(wrapper).toMatchSnapshot();
    });

    it("prop type has specific allowed values", () => {
        console.error = jest.fn();

        const allowedSizes = shallow(
            <div>
                <IconPreview type="material-icons" />
                <IconPreview type="payment-icon" />
                <IconPreview type="flag-icon" />
            </div>
        );

        expect(console.error).not.toHaveBeenCalled();

        const illegalSize = shallow(<IconPreview type="invalid_value" />);

        expect(console.error).toHaveBeenCalled();
        expect(allowedSizes).toMatchSnapshot();
        expect(illegalSize).toMatchSnapshot();
    });

    it("prop size is an enum with specified allowed values", () => {
        console.error = jest.fn();

        const allowedSizes = shallow(
            <div>
                <IconPreview size="tiny" type="material-icons" name="visa" />
                <IconPreview size="small" type="payment-icon" name="amex" />
                <IconPreview size="medium" type="payment-icon" name="diners" />
                <IconPreview size="large" type="payment-icon" name="mastercard" />
                <IconPreview size="huge" type="payment-icon" name="mobilepay" />
            </div>
        );

        expect(console.error).not.toHaveBeenCalled();

        const illegalSize = shallow(<IconPreview name="amex" size="test" type="payment-icon" />);

        expect(console.error).toHaveBeenCalled();

        expect(allowedSizes).toMatchSnapshot();
        expect(illegalSize).toMatchSnapshot();
    });

    it("renders a material icon when type is equal to material-icons", () => {
        const wrapper = shallow(<IconPreview name="android" type="material-icons" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<i className="material-icons" aria-hidden="true">android</i>)).toEqual(true);
    });

    it("renders correct icon when prop name is provided", () => {
        const wrapper = shallow(<IconPreview name="amex" type="payment-icon" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<i className="payment-icon payment-icon-amex" aria-hidden="true"></i>)).toEqual(true);
    });

    it("renders correct size when prop size is provided", () => {
        const wrapper = shallow(<IconPreview name="amex" size="large" type="payment-icon" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<i className="payment-icon payment-icon-large payment-icon-amex" aria-hidden="true"></i>)).toEqual(true);
    });

    it("renders icon with custom when prop className is provided", () => {
        const wrapper = shallow(<IconPreview name="amex" type="payment-icon" className="test test2 test-3" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<i className="payment-icon payment-icon-amex test test2 test-3" aria-hidden="true"></i>)).toEqual(true);
    });

    it("renders a preview wrapper around the <i> tag when prop preview is true", () => {
        const wrapper = shallow(<IconPreview name="amex" type="payment-icon" preview />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(
            <div className="icon-preview">
                <i className="payment-icon payment-icon-amex" aria-hidden="true"></i>
                <code className="code-tags code-tags-secondary mt-2">amex</code>
            </div>
        )).toEqual(true);
    });

    it("renders a preview wrapper around material-icon when prop previewSize is true", () => {
        const wrapper = shallow(<IconPreview name="android" type="material-icons" size="large" previewSize />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(
            <div className="icon-preview">
                <i className="material-icons material-icons-large" aria-hidden="true">android</i>
                <code className="code-tags code-tags-secondary mt-2">large</code>
            </div>
        )).toEqual(true);
    });
});
