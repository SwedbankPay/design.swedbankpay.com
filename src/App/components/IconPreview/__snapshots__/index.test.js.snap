// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IconPreview -  prop name is marked as required 1`] = `
<i
  aria-hidden="true"
  className="payment-icon payment-icon-undefined"
/>
`;

exports[`Component: IconPreview -  prop size is an enum with specified allowed values 1`] = `
<div>
  <IconPreview
    name="visa"
    size="tiny"
    type="material-icons"
  />
  <IconPreview
    name="amex"
    size="small"
    type="payment-icon"
  />
  <IconPreview
    name="diners"
    size="medium"
    type="payment-icon"
  />
  <IconPreview
    name="mastercard"
    size="large"
    type="payment-icon"
  />
  <IconPreview
    name="mobilepay"
    size="huge"
    type="payment-icon"
  />
</div>
`;

exports[`Component: IconPreview -  prop size is an enum with specified allowed values 2`] = `
<i
  aria-hidden="true"
  className="payment-icon payment-icon-test payment-icon-amex"
/>
`;

exports[`Component: IconPreview -  prop type has specific allowed values 1`] = `
<div>
  <IconPreview
    type="material-icons"
  />
  <IconPreview
    type="payment-icon"
  />
  <IconPreview
    type="flag-icon"
  />
</div>
`;

exports[`Component: IconPreview -  prop type has specific allowed values 2`] = `
<i
  aria-hidden="true"
  className="invalid_value invalid_value-undefined"
/>
`;

exports[`Component: IconPreview -  renders 1`] = `
<i
  aria-hidden="true"
  className="payment-icon payment-icon-amex"
/>
`;

exports[`Component: IconPreview -  renders a material icon when type is equal to material-icons 1`] = `
<i
  aria-hidden="true"
  className="material-icons"
>
  android
</i>
`;

exports[`Component: IconPreview -  renders a preview wrapper around material-icon when prop previewSize is true 1`] = `
<div
  className="icon-preview"
>
  <i
    aria-hidden="true"
    className="material-icons material-icons-large"
  >
    android
  </i>
  <code
    className="code-tags code-tags-secondary mt-2"
  >
    large
  </code>
</div>
`;

exports[`Component: IconPreview -  renders a preview wrapper around the <i> tag when prop preview is true 1`] = `
<div
  className="icon-preview"
>
  <i
    aria-hidden="true"
    className="payment-icon payment-icon-amex"
  />
  <code
    className="code-tags code-tags-secondary mt-2"
  >
    amex
  </code>
</div>
`;

exports[`Component: IconPreview -  renders correct icon when prop name is provided 1`] = `
<i
  aria-hidden="true"
  className="payment-icon payment-icon-amex"
/>
`;

exports[`Component: IconPreview -  renders correct size when prop size is provided 1`] = `
<i
  aria-hidden="true"
  className="payment-icon payment-icon-large payment-icon-amex"
/>
`;

exports[`Component: IconPreview -  renders icon with custom when prop className is provided 1`] = `
<i
  aria-hidden="true"
  className="payment-icon payment-icon-amex test test2 test-3"
/>
`;
